errors/errors.go

```diff
@@ -0,0 +1,62 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"runtime"
+)
+
+type Error struct {
+	function FCall
+	Err      error
+	Severity logrus.Level
+}
+func (e Error) Error() string {
+	return e.Err.Error()
+}
+type FCall string
+
+func (o FCall) String() string {
+	return string(o)
+}
+
+func E(functionCall FCall, args ...interface{}) error {
+	e := Error{function: functionCall}
+	if len(args) == 0 {
+		msg := "errors.E called with 0 args"
+		_, file, line, ok := runtime.Caller(1)
+		if ok {
+			msg = fmt.Sprintf("%v - %v:%v", msg, file, line)
+		}
+		e.Err = errors.New(msg)
+	}
+	for _, a := range args {
+		switch a := a.(type) {
+		case error:
+			e.Err = a
```

Comment:
Should we set the Severity here?

Reply:
In same cases, it could be handy to have loglevel as severity, but yeah, there should not be another logging package, either logzero or logrus should be chosen. I will stick with logzero.